File: code/type-coercion.java
String label = "The width is ";
int width = 94;
String widthLabel = label + width;
System.out.println(widthLabel);
// widthLabel is a String

// TypeCoercion Error Example:
System.out.println('A' + 0);
// 65
System.out.println('A' + 50 + "A");
// 115A
System.out.println("A" + 'A' + 50);
// AA50

File: code/type-coercion.kt
val label = "The width is "
val width = 94
val widthLabel = label + width
if(widthLabel is String) {
        println("widthLabel is of type String")
    }
// widthLabel is a String

// TypeCoercion Error Example:
println('A' + 0)
// A
println('A' + 50 + "A")
// sA
println("A" + 'A' + 50)
// AA50
